subcommand/connect-init/command.go

```diff
@@ -140,6 +143,15 @@ func (c *Command) Run(args []string) int {
 		}
 		for _, svc := range serviceList {
 			c.UI.Info(fmt.Sprintf("Registered service has been detected: %s", svc.Service))
+			// When ACLs are enabled: If the flagServiceAccountName is empty, it means the service name pod annotation
+			// was set, so the check for service account name == consul service name has already occurred in
+			// container_init.go. If the flagServiceAccountName is not empty, we need to check whether it matches the
+			// Kubernetes service name.
+			if c.flagACLAuthMethod != "" && c.flagServiceAccountName != "" && svc.Meta[connectinject.MetaKeyKubeServiceName] != c.flagServiceAccountName {
```

Comment:
In general, we should fail if the service account doesn't match the consul service account name rather than Kubernetes. But I guess in this case, consul service name == kubernetes svc name. If it didn't then we would have failed in the webhook. If you don't think we should consolidate these two cases as I've mentioned above, then I think adding a comment here would be nice.